### PR TITLE
refactor(connection): migration connection api to service

### PR DESCRIPTION
**Pull request by oceanlvr**
### PR Checklist

If you have any questions, you can refer to the [Contributing Guide](https://github.com/emqx/MQTTX/blob/master/.github/CONTRIBUTING.md)

#### What is the current behavior?

refactor connection service API

#### What is the new behavior?

Please describe the new behavior or provide screenshots.

#### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Specific Instructions

Are there any specific instructions or things that should be known prior to review?

### CODE REVIEW

File: src/database/services/ConnectionService.ts [hunk 1]
```diff
@@ -2,35 +2,137 @@ import { Service } from 'typedi'
...
+
+  public async loadAllConnectionsIds(): Promise<number[] | undefined> {
+    let res: number[] | undefined = undefined
+    const query: ConnectionEntity[] | undefined = await this.connectionRepository.find({
+      select: ['id'],
+    })
+    if (!query) {
+      return
     }
+    res = query.map((e) => e.id as number)
```
Comment (ysfscream) [inline src/database/services/ConnectionService.ts:135]: Why is it asserted here that the id is number?

File: src/database/services/ConnectionService.ts [hunk 2]
```diff
@@ -2,35 +2,137 @@ import { Service } from 'typedi'
...
+  public async updateConnection(id: number, data: Partial<ConnectionModel>): Promise<ConnectionModel | undefined> {
+    let res: ConnectionEntity | undefined = await this.connectionRepository.findOne(id)
+    if (!res) {
+      return
+    }
+    deepMerge(res, data)
+    const query: ConnectionEntity | undefined = await this.connectionRepository.save(res)
+    return {
+      ...query,
+      isCollection: false,
```
Comment (ysfscream) [inline src/database/services/ConnectionService.ts:44]: Isn't the isCollection field inside ConnectionEntity?

File: src/database/services/ConnectionService.ts [hunk 3]
```diff
@@ -2,35 +2,137 @@ import { Service } from 'typedi'
...
+    } as ConnectionModel
+    return res
+  }
+
+  public async loadAllConnections(): Promise<ConnectionModel[] | undefined> {
+    const query: ConnectionEntity[] | undefined = await this.connectionRepository.find()
+    if (!query) {
+      return
+    }
+    return query.map((e) => {
```
Comment (ysfscream) [inline src/database/services/ConnectionService.ts:95]: It is better to try not to use single letters for parameter naming.

File: src/database/services/ConnectionService.ts (hunk 2 above)
Comment (oceanlvr) [reply src/database/services/ConnectionService.ts:44]: Yes, do we need to define it in ConnectionEntity?

Comment (ysfscream) [reply src/database/services/ConnectionService.ts:44]: This field is valid, right? If it's available, it's still stored in the database.

File: src/database/models/CollectionEntity.ts [hunk 4]
```diff
@@ -7,17 +7,23 @@ export default class CollectionEntity {
...
-  payloadType!: string
-
-  @Column({ type: 'integer', comment: 'order in the collection' })
+  @Column({ type: 'integer', nullable: true, comment: 'order in the collection' })
   orderId!: number
 
-  @OneToMany(() => ConnectionEntity, (connection) => connection.collection)
-  connections!: ConnectionEntity[]
+  @Column({ type: 'boolean', default: true })
+  isCollection?: true
```
Comment (ysfscream) [inline src/database/models/CollectionEntity.ts:16]: true -> boolean?

Comment (oceanlvr) [reply src/database/models/CollectionEntity.ts]: maybe it's better just use true, so that we don't need to handle the type transfer in the services layer

**Review by ysfscream — APPROVED** [review]

**Pull request merged**